### PR TITLE
Jetpack Sync: Add sync event for plugin installation

### PR DESCRIPTION
**Pull request by gititon**
Jetpack Sync: Add sync event for plugin installation

#### Changes proposed in this Pull Request:

* Addition of 'jetpack_installed_plugin' action

#### Testing instructions:

* phpunit

#### Proposed changelog entry for your changes:

Added 'jetpack_installed_plugin' action

### CODE REVIEW

File: sync/class.jetpack-sync-module-plugins.php [hunk 1]
```diff
@@ -24,6 +26,25 @@ public function init_before_send() {
...
+		if ( 'Plugin_Upgrader' === get_class( $upgrader ) ) {
+			$plugin_path = $upgrader->plugin_info();
+			$plugins = get_plugins();
+			$plugin_info = $plugins[ $plugin_path ];
+
+			/**
+			 * Signals to the sync listener that a plugin was installed and a sync action
+			 * reflecting the installation and the plugin info should be sent
+			 *
+			 * @since 4.8.3
```
Comment (enejb) [inline sync/class.jetpack-sync-module-plugins.php]: This will probably land in 4.9.0

File: sync/class.jetpack-sync-module-plugins.php [hunk 2]
```diff
@@ -24,6 +26,25 @@ public function init_before_send() {
 		//Note that we don't simply 'expand_plugin_data' on the 'delete_plugin' action here because the plugin file is deleted when that action finishes
 	}
 
+	public function install_plugin( $upgrader ) {
+		if ( 'Plugin_Upgrader' === get_class( $upgrader ) ) {
```
Comment (enejb) [inline sync/class.jetpack-sync-module-plugins.php]: Do we get any more info then what the upgrader class is to determine what action was take?

Comment (gititon) [inline sync/class.jetpack-sync-module-plugins.php]: Yes, I've updated the code to make use of the additional information.

File: sync/class.jetpack-sync-module-plugins.php (hunk 1 above)
Comment (gititon) [inline sync/class.jetpack-sync-module-plugins.php]: Updated to 4.9.0

**Comment by enejb** [comment]
Thanks for testing the order of things! 
:shipit:

**Pull request merged**